homeassistant/helpers/update_coordinator.py

```diff
@@ -84,9 +84,18 @@ def __init__(
         self.update_interval = update_interval
         self._shutdown_requested = False
         if config_entry is UNDEFINED:
+            # late import to avoid circular imports
+            from . import frame  # pylint: disable=import-outside-toplevel
+
+            frame.report_usage(
+                "rely on the ContextVar, but should passing the config entry explicitly - see "
+                "https://developers.home-assistant.io/blog/xxxxxxx",
+                core_behavior=frame.ReportBehavior.ERROR,
+                custom_integration_behavior=frame.ReportBehavior.LOG,
```

Comment:
<https://github.com/home-assistant/core/pull/127980#issuecomment-2407249077>
>  it's **not wanted** to force custom integrations to change.

Reply:
If we don't want to trouble custom integrations, I think we can use:
```suggestion
                core_behavior=frame.ReportBehavior.ERROR,
                cor_integration_behavior=frame.ReportBehavior.LOG,
                custom_integration_behavior=frame.ReportBehavior.NONE,
```

Reply:
I thought this statements was meant to be valid just for the initial PR, but yes - if we don't want to remove the ability of relying on the ContextVar for custom components, then we need to change this and also remove the "breaks in version" as ludeeus mentioned below 👍